hashicorp/dev-portal
# feat: handle 404 tutorial views

## ✅ "Ready for Review" checklist




---

## 🔗 Relevant links



- [Preview link](https://dev-portal-git-kshandle-404-hashicorp.vercel.app/waypoint/tutorials) 🔎
- [Asana task](https://app.asana.com/0/1199634971449915/1202016460311408) 🎟️

## 🗒️ What

This PR adjusts the tutorial and collection page prop generation functions to return the 404 page if the collection or tutorial doesn't exist. The design implementation of the 404 page will occur later

## 🧪 Testing

- Visit this 404 link for a [tutorial](https://dev-portal-git-kshandle-404-hashicorp.vercel.app/waypoint/tutorials/get-started-docker/fake-tutorial) and [collection](https://dev-portal-git-kshandle-404-hashicorp.vercel.app/waypoint/tutorials/fake-collection) page
- Visit a valid tutorial and collection page, make sure they work as expected.
